### PR TITLE
Update OrtExtensionsUsage to also use the ORT Objective-C API.

### PR DESCRIPTION
**Pull request by edgchen1**
Note: This PR depends on the new Objective-C API added in https://github.com/microsoft/onnxruntime/pull/16603.

### CODE REVIEW

File: test/ios/OrtExtensionsUsage/Podfile [hunk 1]
```diff
@@ -1,11 +1,13 @@
-platform :ios, '11.0'
+platform :ios, '12.0'
 
 target 'OrtExtensionsUsage' do
   # Comment the next line if you don't want to use dynamic frameworks
   use_frameworks!
 
   # Pods for OrtExtensionsUsage
-  pod 'onnxruntime-c'
+  #pod 'onnxruntime-objc'  # can use this after 1.16 is released
```
Comment (edgchen1) [inline test/ios/OrtExtensionsUsage/Podfile:8]: will need to use a pre-release version prior to 1.16, after the corresponding change in ORT to add the custom op registration API gets in.

**Comment by edgchen1** [comment]
some formatting changes - suggest ignoring whitespace changes when viewing diff

File: test/ios/OrtExtensionsUsage/OrtExtensionsUsage/OrtSwiftClient.swift [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    throw OrtSwiftClientError.error("Unexpected output shape")
+  }
+
+  let expected_output_element_type = ORTTensorElementDataType.uInt8
+  guard output_type_and_shape.elementType == expected_output_element_type else {
+    throw OrtSwiftClientError.error("Unexpected output element type")
+  }
+
+  let output_data: Data = try output_tensor.tensorData() as Data
+  let expected_pixel_bgr_data: [UInt8] = [128, 64, 32]
```
Comment (skottmckay) [inline test/ios/OrtExtensionsUsage/OrtExtensionsUsage/OrtSwiftClient.swift:59]: nit: A comment connecting the 128/64/32 values to the ordering (BGR and channels last) and the model (which I assume has 32 for all R values, 64 for all G values and 128 for all B values based on the model name) would save the reader from having to figure that slightly indirect relationship out.

**Review by skottmckay — APPROVED** [review]

File: test/ios/OrtExtensionsUsage/Podfile [hunk 3]
```diff
@@ -5,7 +5,7 @@ target 'OrtExtensionsUsage' do
   use_frameworks!
 
   # Pods for OrtExtensionsUsage
-  pod 'onnxruntime-c'
+  pod 'onnxruntime-objc'
```
Comment (wenbingl) [inline test/ios/OrtExtensionsUsage/Podfile:8]: Is there any version restriction here, something like onnxruntime-objc >= 1.16?

Comment (edgchen1) [reply test/ios/OrtExtensionsUsage/Podfile]: yes, 1.16 is the minimum required version now. it should get the latest like this, but I added `>= 1.16.0`.

**Review by skottmckay — APPROVED** [review]
:shipit:

**Pull request merged**